### ISSUE TITLE
Warnings after rolling upgrade

### ISSUE THREAD

**Issue by leshadubenyuk** [dotnet/orleans#6574]
Hey! 
After rolling upgrades of the orleans application i noticed the some warnings in logs tha occures every 5 minutes. After some investigation, it seems like some grain doesnt get the information about another silo upgrade, because target in log message uses old generation silo number. Nevertheless, its not an issue because target seems to resolve itself. So i assume its not critical. I am just wondering why they appearing, perhaps some mechanism inside orleans doesnt work as expected.

Log message:
```
Microsoft.Orleans.Networking - Attempting to send message addressed to S127.25.53.6:22222:328876289 to connection with S127.25.53.6:22222:328270038. 
Message Response S127.25.52.94:22222:328876233*stg/15/0000000f@S0000000f->S127.25.53.6:22222:328876289*stg/15/0000000f@S0000000f
```

**Comment by sergeybykov** [comment]
What version is this?

**Comment by leshadubenyuk** [comment]
Orleans version used 3.0.2.

**Comment by sergeybykov** [comment]
I'm pretty sure this has been fixed since. Please upgrade to 3.1.7 or 3.2.0.

**Comment by leshadubenyuk** [comment]
Issue resolved after upgrading to 3.2.0, many thanks!
